core-v1/src/main/scala/dsl/SelectElement.scala

```diff
@@ -0,0 +1,29 @@
+package com.rallyhealth.vapors.v1
+
+package dsl
+
+import shapeless.Id
+
+import scala.annotation.implicitNotFound
+
+// TODO: Include OP type here?
+// TODO: Figure out how to make W invariant?
+trait SelectElement[W[+_]] {
```

Comment:
I should document this. I don't think making `W` invariant on its parameter is necessary or really preferable, so I should remove that TODO. I played around with adding the `OP`, but I'm not sure if it is necessary. Might be worth another shot though.